zcncore/execute_transactions.go

```diff
@@ -86,14 +116,34 @@ func MinerSCKill(providerId string, providerType Provider, client ...string) (ha
 
 }
 
-func StorageSCCollectReward(providerId string, providerType Provider, client ...string) (hash, out string, nonce int64, txn *transaction.Transaction, err error) {
-	return transaction.SmartContractTxn(StorageSmartContractAddress, transaction.SmartContractTxnData{
+func StorageSCStakePoolLock(providerId string, providerType Provider, lock uint64, fee uint64, client ...string) (hash, out string, nonce int64, txn *transaction.Transaction, err error) {
```

Comment:
You can use zboxcore function : func StakePoolLock(providerType ProviderType, providerID string, value, fee uint64) (hash string, nonce int64, err error). 

Please avoid duplicates.

Reply:
Don't recreate it independently. You can build on the exisiting function so that updates to them are made at the same time.